rubin_sim/maf/batches/radar_limited.py

```diff
@@ -0,0 +1,1079 @@
+__all__ = ("radar_limited",)
+
+import astropy.units as u
+import healpy as hp
+import numpy as np
+
+import rubin_sim.maf as maf
+import rubin_sim.maf.maps as maps
+import rubin_sim.maf.metric_bundles as mb
+import rubin_sim.maf.metrics as metrics
+import rubin_sim.maf.plots as plots
+import rubin_sim.maf.slicers as slicers
+
+from .common import extended_summary, filter_list, lightcurve_summary, standard_summary
+
+
+def radar_limited(
+    runName="run name",
+    nside=64,
+    benchmarkArea=18000,
+    benchmarkNvisits=825,
+    minNvisits=750,
+    long_microlensing=True,
+    srd_only=False,
+    mjd0=None,
+):
+    """A batch of metrics for looking at survey performance relative to the SRD and the main
+    science drivers of LSST.
+
+    Parameters
+    ----------
+    long_microlensing : `bool` (True)
+        Add the longer running microlensing metrics to the batch (subset of crossing times only)
+    srd_only : `bool` (False)
+        Only return the SRD metrics
+    """
+
+    bundleList = []
+    # Get some standard per-filter coloring and sql constraints
+    filterlist, colors, filterorders, filtersqls, filterinfo_label = filter_list(all=False)
+    (
+        allfilterlist,
+        allcolors,
+        allfilterorders,
+        allfiltersqls,
+        allfilterinfo_label,
+    ) = filter_list(all=True)
+
+    standardStats = standard_summary(withCount=False)
+
+    # This is the default slicer for most purposes in this batch. Note that the cache is off -
+    # if the metric requires a dust map, this is not the right slicer to use.
+    healpixslicer = slicers.HealpixSlicer(nside=nside, use_cache=True)
+    subsetPlots = [plots.HealpixSkyMap(), plots.HealpixHistogram()]
+
```

Comment:
Stylistically, the code could be clearer and more reusable if the different sections (marked off with `##########` blocks) were extracted into separate functions, and the `radar_limited` function just call each in turn.

Reply:
Normally I agree. In this case the code is generating output for MAF, so I really like having all the code that generates numbers and plots for a web page in one place. Otherwise when someone says "change this colorbar to log", or "was this configured for 3 or 4 points before the peak of the light curve" it becomes a nightmare to trace things back to the source.